### PR TITLE
feat(pnpm): add support for pnpm 7

### PR DESCRIPTION
**Pull request by GermainBergeron**
Pnpm version 7 was released a few weeks ago and we realized it was breaking the releases in [Plasma](https://github.com/coveo/plasma/). I added some logic to add the `--` for pnpm 6 or remove them for pnpm 7.

I also added unit tests

### CODE REVIEW

File: src/pnpm/utils/runOnChanged.ts [hunk 1]
```diff
@@ -9,13 +9,15 @@ export const runOnChanged = (
   forcePackages: string[],
   excludePackages: string[]
 ) => {
+  const version = spawnSync("pnpm", ["--version"], pnpmLogger).stdout.trim();
```
Comment (louis-bompart) [inline src/pnpm/utils/runOnChanged.ts:12]: Can you use `semver` to coerce the version?
It'll be more reliable I think

**Review by louis-bompart — COMMENTED** [review]
LGTM, do we want to support the previous version of PNPM tho?

**Review by nathanlb — APPROVED** [review]
Dunno much about how this project works but the diff looks sensible to me.

**Review by aiheon — APPROVED** [review]

**Comment by louis-bompart** [comment]
> > LGTM, do we want to support the previous version of PNPM tho?
> 
> Version 7 was quite breaking and is still early that's why I kept support for version 6 (also it was simple), but I hesitated. Since this project is early and the PNPM part probably just used by my team we could just scrap support for v6. What's your thought on that?

We can remove V6 support for the v1, that should leave about a month for pnpm 7 to stabilize

**Review by louis-bompart — APPROVED** [review]

**Pull request merged**